plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb.test.ts

```diff
@@ -244,6 +249,93 @@ describe('MongoDBInstrumentation', () => {
     });
   });
 
+  describe('when specifying a responseHook configuration', () => {
+    const dataAttributeName = 'mongodb_data';
+    beforeEach(() => {
+      memoryExporter.reset();
+    });
+
+    describe('with a valid function', () => {
+      beforeEach(() => {
+        create({
+          responseHook: (span: Span, result: CommandResult) => {
+            span.setAttribute(dataAttributeName, JSON.stringify(result.result));
+          },
+        });
+      });
+
+      it('should attach response hook data to the resulting span for insert function', done => {
+        const insertData = [{ a: 1 }, { a: 2 }, { a: 3 }];
+        const span = provider.getTracer('default').startSpan('insertRootSpan');
+        context.with(trace.setSpan(context.active(), span), () => {
+          collection.insertMany(insertData, (err, result) => {
+            span.end();
+            assert.ifError(err);
+            const spans = memoryExporter.getFinishedSpans();
+            const insertSpan = spans[0];
+
+            assert.strictEqual(
+              insertSpan.attributes[dataAttributeName],
+              JSON.stringify(result.result, Object.keys(result.result).sort())
```

Comment:
I am probably missing something, but the hook is storing the value of `JSON.stringify(result.result)`, but the the test is comparing it to `JSON.stringify(result.result, Object.keys(result.result).sort())`. What is the motivation for the second parameter to `stringify`?

Reply:
the object attributes lose their sorting - so i have to rearrange them for the test.

Reply:
In the hook you call `JSON.stringify` as well without sorting, not sure how it is different

Reply:
`SpanAttributeValue` cannot not receive an object:
```
export declare type SpanAttributeValue = string | number | boolean | Array<null | undefined | string> | Array<null | undefined | number> | Array<null | undefined | boolean>;
```
that is why i'm using `JSON.stringify`.
when i compare it with the real result (**which is indeed an object!!**) i have to sort it for the test to pass.
`JSON.parse` is not applicable since `insertSpan.attributes[dataAttributeName]` might be undefined.

Reply:
mmm, it's really a minor, not sure it's worth the time to discuss, but I still can't understand how come you take the exact same object, apply the exact same function `JSON.stringify` on it twice (in the hook and in the assert), and you have to use the sort only in one place.

Regarding `insertSpan.attributes[dataAttributeName]`, it cannot be undefined, as this is exactly what the test is asserting. If it is undefined the test should fail.

But anyhow, It's not that important I guess. So long as it works.

Reply:
Would it not be easier to check `assert.deepStrictEqual(JSON.parse(insertSpan.attributes[dataAttributeName]), result.result)`? This sidesteps the ordering part altogether

Reply:
`SpanAttributeValue` by definition can be undefined... I can `JSON.parse` it but i'll have to add `|| ''` which I find uglier than the current solution.
And as you mentioned - if it is undefined, then this test will fail - so we're covered.

Reply:
@dyladan @blumamir 
Changing to 
```
assert.deepStrictEqual(
              JSON.parse((insertSpan.attributes[dataAttributeName] || '').toString()),
              result.result
            );
```
Also in the test below.
Thanks!

Reply:
Thanks :)
Will it also work with `JSON.parse(insertSpan.attributes[dataAttributeName] as string)`?

Reply:
Working as well ;)
Fixed